cns/service/main.go

```diff
@@ -651,16 +651,14 @@ func main() {
 		return
 	}
 
+	// copy ChannelMode from cnsconfig to HTTPRemoteRestService config
+	config.ChannelMode = cnsconfig.ChannelMode
 	if cnsconfig.ChannelMode == cns.Managed {
-		config.ChannelMode = cns.Managed
 		privateEndpoint = cnsconfig.ManagedSettings.PrivateEndpoint
 		infravnet = cnsconfig.ManagedSettings.InfrastructureNetworkID
 		nodeID = cnsconfig.ManagedSettings.NodeID
-	} else if cnsconfig.ChannelMode == cns.CRD {
-		config.ChannelMode = cns.CRD
-	} else if cnsconfig.ChannelMode == cns.MultiTenantCRD {
-		config.ChannelMode = cns.MultiTenantCRD
-	} else if acn.GetArg(acn.OptManaged).(bool) {
+	}
+	if acn.GetArg(acn.OptManaged).(bool) {
```

Comment:
Not crazy about unsafe type assertions outside of tests. This removes the lurking panic:
```suggestion
	if isManaged, ok := acn.GetArg(acn.OptManaged).(bool); ok && isManaged {
```

Reply:
makes sense to me, applied the suggestion